api/v1alpha1/extendeddaemonset_types.go

```diff
@@ -99,22 +99,32 @@ type ExtendedDaemonSetSpecStrategyCanary struct {
 // ExtendedDaemonSetSpecStrategyCanaryAutoPause defines the canary deployment AutoPause parameters of the ExtendedDaemonSet.
 // +k8s:openapi-gen=true
 type ExtendedDaemonSetSpecStrategyCanaryAutoPause struct {
+	// Enabled enables AutoPause.
+	// Default value is true.
 	Enabled *bool `json:"enabled,omitempty"`
 	// MaxRestarts defines the number of tolerable (per pod) Canary pod restarts after which the Canary deployment is autopaused.
+	// Default value is 2.
 	MaxRestarts *int32 `json:"maxRestarts,omitempty"`
-	// MaxSlowStartDuration defines the maximum slow start duration for a pod (stuck in Creating state) after which the.
-	// Canary deployment is autopaused
+	// MaxSlowStartDuration defines the maximum slow start duration for a pod (stuck in Creating state) after which the Canary deployment is autopaused.
+	// There is no default value.
 	MaxSlowStartDuration *metav1.Duration `json:"maxSlowStartDuration,omitempty"`
 }
 
 // ExtendedDaemonSetSpecStrategyCanaryAutoFail defines the canary deployment AutoFail parameters of the ExtendedDaemonSet.
 // +k8s:openapi-gen=true
 type ExtendedDaemonSetSpecStrategyCanaryAutoFail struct {
+	// Enabled enables AutoFail.
+	// Default value is true.
 	Enabled *bool `json:"enabled,omitempty"`
 	// MaxRestarts defines the number of tolerable (per pod) Canary pod restarts after which the Canary deployment is autofailed.
+	// Default value is 5.
 	MaxRestarts *int32 `json:"maxRestarts,omitempty"`
 	// MaxRestartsDuration defines the maximum duration of tolerable Canary pod restarts after which the Canary deployment is autofailed.
+	// There is no default value.
 	MaxRestartsDuration *metav1.Duration `json:"maxRestartsDuration,omitempty"`
+	// MaxDuration defines the maximum duration of a Canary after which the Canary deployment is autofailed. This is a safeguard against lengthy Canary pauses.
+	// There is no default value.
+	MaxDuration *metav1.Duration `json:"maxDuration,omitempty"`
```

Comment:
```suggestion
	MaxCanaryDuration *metav1.Duration `json:"maxDuration,omitempty"`
```

No strong opinion, it's just that the other option `MaxRestartsDuration` makes `MaxDuration` a bit confusing. `MaxCanaryDuration` would clearer imho.